### PR TITLE
Fix `is_boolean`

### LINKED ISSUE #11: Non-boolean field producing a boolean summary table
When running `dataset-report` (`v0.0.9`) on bmi-short-data-report ([repo](https://github.com/opensafely/bmi-short-data-report)/[job](https://jobs.opensafely.org/datalab/bmi-short-data-report/bmi-short-data-report/8048/cj5oxcpdscvmfon5/)), `dereg_date` (which is of the `datetime64[ns]` data type) shows up in the list of tables that should currently only display summaries for `bool` data types. It is blank for both the `Count` and `Percentage` columns, but it should not have a table at all.

### PR DESCRIPTION
**Pull request by iaindillingham**
I think it's best to read this PR commit-by-commit, because only 0530c7f is the fix. The first four enhance the tests (d017476, 3e5aae7, d307396, 960f86a). The next three add new tests (cfaf961, d0d48db, ed13780).

Fixes #11

### CODE REVIEW

File: tests/test_dataset_report.py [hunk 1]
```diff
@@ -35,24 +35,38 @@ def test_get_name(path, name):
...
+    def test_with_boolean_values(self, data, dtype):
+        assert dataset_report.is_boolean(pandas.Series(data, dtype=dtype))
+
+    @pytest.mark.parametrize(
+        "data,dtype",
+        [
+            ([0, 2], int),
+            ([0.1, 0.2], float),
+            ([numpy.nan, 2], float),
+            (["0", "1"], str),
```
Comment (robinyjpark) [inline tests/test_dataset_report.py:61]: Am I understanding correctly that we _won't_ be consider ["0", "1"] as bool? Should we want to? I am trying to think of a case where we have "0" and "1" as the unique values in a series, and it doesn't serve as an indicator/effectively a bool.

File: tests/test_dataset_report.py [hunk 2]
```diff
@@ -35,24 +35,38 @@ def test_get_name(path, name):
...
+            ([datetime(2022, 1, 1), datetime(2022, 1, 2)], "datetime64[ns]"),
+            (["0", "1"], "category"),
```
Comment (robinyjpark) [inline tests/test_dataset_report.py:63]: Same comment as above

**Review by robinyjpark — COMMENTED** [review]
Just a small question with regards to ["0","1"] string/category (more in comments in line)

File: tests/test_dataset_report.py (hunk 1 above)
Comment (iaindillingham) [reply tests/test_dataset_report.py:61]: That's correct, and I don't think we want to 🙂. We know that a boolean series can be represented as a numeric series, with values drawn from the set `{0, 1, numpy.nan}`. A case where a boolean series is represented as a string (`object`) series would be unusual. I'd prefer us not to make any assumptions about one until we see one.

Comment (iaindillingham) [reply tests/test_dataset_report.py:61]: Ultimately, we'd probably want to perform the same operations on a numeric-as-boolean as a string-as-boolean. Keeping the two separated, however, hopefully emphasizes that the former is usual, but the latter is unusual.

Comment (robinyjpark) [reply tests/test_dataset_report.py:61]: Got it, that makes sense. Then I am happy to approve!

**Pull request merged**